Search.setIndex({"docnames": ["api", "generated/genecomb", "index", "usage"], "filenames": ["api.rst", "generated/genecomb.rst", "index.rst", "usage.rst"], "titles": ["API", "genecomb", "Welcome to genecomb\u2019s documentation!", "Usage"], "terms": {"index": [2, 3], "modul": 2, "search": 2, "page": 2, "usag": 2, "instal": 2, "To": 3, "us": 3, "genecomb": [], "first": 3, "pip": 3, "venv": 3, "api": 2, "count": 2, "base": 2, "sequenc": 2, "you": 3, "can": 3, "base_count": [2, 3], "function": [1, 3], "autofunct": [], "class": 1, "test": [], "perform": [], "analysi": [], "auto": [], "read_fasta": [2, 3], "filenam": [], "read": 2, "fasta": 2, "file": 2, "return": 3, "list": 3, "object": 3, "each": 3, "correspond": 3, "self": 3, "thi": 3, "all": 2, "dictionari": 3, "ex": 3, "acgggtac": 3, "A": 3, "2": 3, "c": 3, "g": 3, "3": 3, "t": 3, "1": 3, "x": 3, "0": 3, "The": 3, "rtype": [], "dict": 3, "type": 3, "calcul": 2, "gc": 2, "gc_content": [2, 3], "i": 3, "percentag": 3, "": 3, "nucleotid": 2, "If": 3, "100": 3, "long": 3, "have": 3, "20": 3, "5": 3, "your": 3, "25": 3, "float": 3, "find": 2, "number": 2, "non": 2, "non_nucleotide_count": [2, 3], "palindrom": 2, "subsequ": 2, "find_palindrom": [2, 3], "It": 3, "help": 3, "know": 3, "ar": 3, "valid": 3, "quickli": 3, "valu": 3, "pars": 3, "locat": 3, "acgt": 3, "length": 3, "unknown": 3, "thei": 3, "consecut": 3, "acngggnnntac": 3, "6": 3, "8": 3, "form": 3, "posit": 3, "In": [], "gene": [], "nucleic": 3, "acid": 3, "doubl": 3, "strand": 3, "dna": 3, "rna": 2, "molecul": 3, "wherebi": 3, "certain": 3, "direct": 3, "e": 3, "one": 3, "ident": 3, "same": 3, "complementari": 3, "sourc": 3, "http": [], "en": [], "wikipedia": [], "org": [], "wiki": [], "palindromic_sequ": [], "removeoverlap": 3, "fals": 3, "within": 3, "seq": 3, "paramet": 3, "str": 3, "entri": 3, "anoth": 3, "start": 3, "end": 3, "_": [], "write": 2, "write_to_fasta": [], "write_to_fasta_fil": [2, 3], "filepath": 3, "append": 3, "line_length": 3, "80": 3, "header": 3, "format": 3, "option": 3, "an": 3, "exist": 3, "overwrit": 3, "default": 3, "param": 3, "line": 3, "when": 3, "standard": 3, "kind": [], "bool": 3, "int": 3, "except": 1, "from": 2, "multipl": 3, "either": 3, "singl": 3, "more": 3, "than": 3, "found": 3, "point": 2, "mutat": 2, "between": 2, "two": 2, "count_point_mut": [2, 3], "should": 3, "mismatch": 3, "occur": 3, "due": 3, "equal": 3, "gene_a": 3, "gene_b": 3, "compar": 3, "b": 3, "rais": 3, "sequencelengthmismatchexcept": 3, "differ": 3, "translat": 2, "protein": 2, "translate_to_protein": [2, 3], "often": 3, "done": 3, "transcrib": 2, "amino": 3, "carri": 3, "out": 3, "tranlsat": 3, "until": 3, "u": 3, "design": 3, "get_rna_transcript": [2, 3], "replac": 3, "zerodivisionerror": 3, "empti": 3}, "objects": {"": [[1, 0, 0, "-", "genecomb"]], "genecomb.GeneComb": [[3, 1, 1, "", "base_counter"], [3, 1, 1, "", "find_palindromes"], [3, 1, 1, "", "gc_content"], [3, 1, 1, "", "get_rna_transcription"], [3, 1, 1, "", "non_nucleotide_counter"], [3, 1, 1, "", "translate_to_protein"], [3, 1, 1, "", "write_to_fasta_file"]], "genecomb": [[3, 1, 1, "", "count_point_mutations"], [3, 1, 1, "", "read_fasta"]]}, "objtypes": {"0": "py:module", "1": "py:function"}, "objnames": {"0": ["py", "module", "Python module"], "1": ["py", "function", "Python function"]}, "titleterms": {"welcom": 2, "genecomb": [1, 2, 3], "": 2, "document": 2, "indic": 2, "tabl": 2, "content": [2, 3], "usag": 3, "instal": 3, "api": 0, "count": 3, "base": 3, "test": [], "calcul": 3, "gc": 3, "find": 3, "number": 3, "non": 3, "nucleotid": 3, "all": 3, "palindrom": 3, "subsequ": 3, "write": 3, "sequenc": 3, "fasta": 3, "file": 3, "read": 3, "from": 3, "point": 3, "mutat": 3, "between": 3, "two": 3, "translat": 3, "protein": 3, "transcrib": 3, "rna": 3}, "envversion": {"sphinx.domains.c": 2, "sphinx.domains.changeset": 1, "sphinx.domains.citation": 1, "sphinx.domains.cpp": 8, "sphinx.domains.index": 1, "sphinx.domains.javascript": 2, "sphinx.domains.math": 2, "sphinx.domains.python": 3, "sphinx.domains.rst": 2, "sphinx.domains.std": 2, "sphinx": 57}, "alltitles": {"Welcome to genecomb\u2019s documentation!": [[2, "welcome-to-genecomb-s-documentation"]], "Indices and tables": [[2, "indices-and-tables"]], "Contents": [[2, "contents"]], "API": [[0, "api"]], "genecomb": [[1, "module-genecomb"]], "Usage": [[3, "usage"]], "Installation": [[3, "installation"]], "Counting Bases": [[3, "counting-bases"]], "Calculating GC Content": [[3, "calculating-gc-content"]], "Finding the Number of Non-nucleotide Bases": [[3, "finding-the-number-of-non-nucleotide-bases"]], "Finding all palindromic subsequences": [[3, "finding-all-palindromic-subsequences"]], "Transcribing a nucleotide sequence to RNA": [[3, "transcribing-a-nucleotide-sequence-to-rna"]], "Translating a nucleotide sequence to protein bases": [[3, "translating-a-nucleotide-sequence-to-protein-bases"]], "Counting Point Mutations between two sequences": [[3, "counting-point-mutations-between-two-sequences"]], "Writing a GeneComb sequence to a FASTA file": [[3, "writing-a-genecomb-sequence-to-a-fasta-file"]], "Reading GeneComb sequences from a FASTA file": [[3, "reading-genecomb-sequences-from-a-fasta-file"]]}, "indexentries": {"genecomb": [[1, "module-genecomb"]], "module": [[1, "module-genecomb"]], "base_counter() (in module genecomb.genecomb)": [[3, "genecomb.GeneComb.base_counter"]], "count_point_mutations() (in module genecomb)": [[3, "genecomb.count_point_mutations"]], "find_palindromes() (in module genecomb.genecomb)": [[3, "genecomb.GeneComb.find_palindromes"]], "gc_content() (in module genecomb.genecomb)": [[3, "genecomb.GeneComb.gc_content"]], "get_rna_transcription() (in module genecomb.genecomb)": [[3, "genecomb.GeneComb.get_rna_transcription"]], "non_nucleotide_counter() (in module genecomb.genecomb)": [[3, "genecomb.GeneComb.non_nucleotide_counter"]], "read_fasta() (in module genecomb)": [[3, "genecomb.read_fasta"]], "translate_to_protein() (in module genecomb.genecomb)": [[3, "genecomb.GeneComb.translate_to_protein"]], "write_to_fasta_file() (in module genecomb.genecomb)": [[3, "genecomb.GeneComb.write_to_fasta_file"]]}})