Search.setIndex({"docnames": ["api", "index", "usage"], "filenames": ["api.rst", "index.rst", "usage.rst"], "titles": ["API", "Welcome to genecomb\u2019s documentation!", "Usage"], "terms": {"index": [0, 1, 2], "modul": 1, "search": 1, "page": 1, "usag": 1, "instal": 1, "To": 2, "us": [0, 2], "genecomb": [], "first": 2, "pip": 2, "venv": 2, "api": 1, "count": [0, 1], "base": [0, 1], "sequenc": [0, 1], "you": [0, 2], "can": 2, "base_count": [0, 1, 2], "function": [0, 2], "autofunct": [], "class": 0, "test": [], "perform": 0, "analysi": 0, "auto": [], "read_fasta": [1, 2], "filenam": [], "read": 1, "fasta": [0, 1], "file": [0, 1], "return": [0, 2], "list": [0, 2], "object": 2, "each": [0, 2], "correspond": [0, 2], "self": 2, "thi": [0, 2], "all": [0, 1], "dictionari": [0, 2], "ex": [0, 2], "acgggtac": [0, 2], "A": [0, 2], "2": [0, 2], "c": [0, 2], "g": [0, 2], "3": [0, 2], "t": [0, 2], "1": [0, 2], "x": [0, 2], "0": [0, 2], "The": [0, 2], "rtype": [], "dict": [0, 2], "type": [0, 2], "calcul": 1, "gc": [0, 1], "gc_content": [0, 1, 2], "i": [0, 2], "percentag": 2, "": [0, 2], "nucleotid": [0, 1], "If": [0, 2], "100": [0, 2], "long": [0, 2], "have": [0, 2], "20": [0, 2], "5": [0, 2], "your": [0, 2], "25": [0, 2], "float": [0, 2], "find": [0, 1], "number": 1, "non": [0, 1], "non_nucleotide_count": [0, 1, 2], "palindrom": [0, 1], "subsequ": 1, "find_palindrom": [0, 1, 2], "It": [0, 2], "help": 2, "know": 2, "ar": [0, 2], "valid": [0, 2], "quickli": 2, "valu": 2, "pars": [0, 2], "locat": [0, 2], "acgt": [0, 2], "length": [0, 2], "unknown": [0, 2], "thei": [0, 2], "consecut": [0, 2], "acngggnnntac": [0, 2], "6": [0, 2], "8": [0, 2], "form": [0, 2], "posit": [0, 2], "In": [], "gene": 0, "nucleic": 2, "acid": [0, 2], "doubl": 2, "strand": 2, "dna": 2, "rna": [0, 1], "molecul": 2, "wherebi": 2, "certain": 2, "direct": 2, "e": 2, "one": [0, 2], "ident": 2, "same": 2, "complementari": 2, "sourc": 2, "http": [], "en": [], "wikipedia": [], "org": [], "wiki": [], "palindromic_sequ": [], "removeoverlap": [0, 2], "fals": [0, 2], "within": [0, 2], "seq": [0, 2], "paramet": [0, 2], "str": [0, 2], "entri": [0, 2], "anoth": [0, 2], "start": [0, 2], "end": [0, 2], "_": [], "write": [0, 1], "write_to_fasta": [], "write_to_fasta_fil": [0, 1, 2], "filepath": [0, 2], "append": [0, 2], "line_length": [0, 2], "80": [0, 2], "header": [0, 2], "format": [0, 2], "option": [0, 2], "an": [0, 2], "exist": [0, 2], "overwrit": [0, 2], "default": [0, 2], "param": [0, 2], "line": [0, 2], "when": [0, 2], "standard": [0, 2], "kind": [], "bool": [0, 2], "int": [0, 2], "except": 0, "from": [0, 1], "multipl": 2, "either": 2, "singl": 2, "more": 2, "than": 2, "found": 2, "point": [0, 1], "mutat": 1, "between": 1, "two": 1, "count_point_mut": [1, 2], "should": 2, "mismatch": 2, "occur": 2, "due": 2, "equal": 2, "gene_a": 2, "gene_b": 2, "compar": 2, "b": 2, "rais": [0, 2], "sequencelengthmismatchexcept": 2, "differ": 2, "translat": [0, 1], "protein": [0, 1], "translate_to_protein": [0, 1, 2], "often": 2, "done": 2, "transcrib": [0, 1], "amino": [0, 2], "carri": [0, 2], "out": [0, 2], "tranlsat": [0, 2], "until": [0, 2], "u": [0, 2], "design": [0, 2], "get_rna_transcript": [0, 1, 2], "replac": [0, 2], "zerodivisionerror": [0, 2], "empti": [0, 2], "descript": 0, "content": 0, "get_reverse_compli": 0, "revers": 0, "complement": 0, "get_compli": 0, "charact": 0, "complimentari": 0, "compliment": 0, "other": 0, "also": 0, "pass": 0, "is_valid_rna_nucleotid": 0, "true": 0, "That": 0}, "objects": {"": [[0, 0, 0, "-", "genecomb"]], "genecomb": [[0, 1, 1, "", "GeneComb"], [2, 2, 1, "", "count_point_mutations"], [2, 2, 1, "", "read_fasta"]], "genecomb.GeneComb": [[2, 2, 1, "", "find_palindromes"], [2, 2, 1, "", "get_rna_transcription"], [2, 2, 1, "", "translate_to_protein"], [2, 2, 1, "", "write_to_fasta_file"]]}, "objtypes": {"0": "py:module", "1": "py:class", "2": "py:function"}, "objnames": {"0": ["py", "module", "Python module"], "1": ["py", "class", "Python class"], "2": ["py", "function", "Python function"]}, "titleterms": {"welcom": 1, "genecomb": [0, 1, 2], "": 1, "document": 1, "indic": 1, "tabl": 1, "content": [1, 2], "usag": 2, "instal": 2, "api": 0, "count": 2, "base": 2, "test": [], "calcul": 2, "gc": 2, "find": 2, "number": 2, "non": 2, "nucleotid": 2, "all": 2, "palindrom": 2, "subsequ": 2, "write": 2, "sequenc": 2, "fasta": 2, "file": 2, "read": 2, "from": 2, "point": 2, "mutat": 2, "between": 2, "two": 2, "translat": 2, "protein": 2, "transcrib": 2, "rna": 2}, "envversion": {"sphinx.domains.c": 2, "sphinx.domains.changeset": 1, "sphinx.domains.citation": 1, "sphinx.domains.cpp": 8, "sphinx.domains.index": 1, "sphinx.domains.javascript": 2, "sphinx.domains.math": 2, "sphinx.domains.python": 3, "sphinx.domains.rst": 2, "sphinx.domains.std": 2, "sphinx": 57}, "alltitles": {"Welcome to genecomb\u2019s documentation!": [[1, "welcome-to-genecomb-s-documentation"]], "Indices and tables": [[1, "indices-and-tables"]], "Contents": [[1, "contents"]], "Usage": [[2, "usage"]], "Installation": [[2, "installation"]], "Counting Bases": [[2, "counting-bases"]], "Calculating GC Content": [[2, "calculating-gc-content"]], "Finding the Number of Non-nucleotide Bases": [[2, "finding-the-number-of-non-nucleotide-bases"]], "Finding all palindromic subsequences": [[2, "finding-all-palindromic-subsequences"]], "Transcribing a nucleotide sequence to RNA": [[2, "transcribing-a-nucleotide-sequence-to-rna"]], "Translating a nucleotide sequence to protein bases": [[2, "translating-a-nucleotide-sequence-to-protein-bases"]], "Counting Point Mutations between two sequences": [[2, "counting-point-mutations-between-two-sequences"]], "Writing a GeneComb sequence to a FASTA file": [[2, "writing-a-genecomb-sequence-to-a-fasta-file"]], "Reading GeneComb sequences from a FASTA file": [[2, "reading-genecomb-sequences-from-a-fasta-file"]], "API": [[0, "api"]], "genecomb": [[0, "module-genecomb"]], "GeneComb": [[0, "id1"]]}, "indexentries": {"genecomb (class in genecomb)": [[0, "genecomb.GeneComb"]], "genecomb": [[0, "module-genecomb"]], "module": [[0, "module-genecomb"]]}})